{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "4bc5bc6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "def getMotifCount(peakFile, genome, motifFile, outputDirectory=\"\"):\n",
    "    \n",
    "    if not os.path.exists(peakFile):\n",
    "        print('Path to peak file is incorrect or does not exist\\n')\n",
    "        return 1\n",
    "\n",
    "    if not os.path.exists(motifFile):\n",
    "        print('Path to motifs is incorrect or does not exist\\n')\n",
    "        return 1\n",
    "    \n",
    "    ## Read in motifFile containing path to .motif files\n",
    "    motifs = pd.read_table(motifFile, sep=\"\\n\", header=None)\n",
    "    motifs = motifs.to_numpy()\n",
    "    \n",
    "    ## Create a string combinining all the .motif paths\n",
    "#     motifString = \"\"\n",
    "#     numOfMotif = motifs.size\n",
    "#     for i in motifs:\n",
    "#         motifString = motifString + i[0] + \" \"\n",
    "        \n",
    "    ## Output File\n",
    "    motifCount = outputDirectory + \"motifCount.txt\"\n",
    "    \n",
    "    ## Call HOMER annotatePeaks to get motif counts\n",
    "    print('annotatePeaks.pl ' + peakFile + \" \" + genome + ' -m ' + motifFile + '-nmotifs > '+ motifCount)\n",
    "    os.system('annotatePeaks.pl ' + peakFile + \" \" + genome + ' -m ' + motifFile + ' -nmotifs > '+ motifCount)\n",
    "    \n",
    "    # Read in HOMER output\n",
    "    motifCountTable = pd.read_table(motifCount, sep=\"\\t\")\n",
    "    os.system('rm ' + motifCount)\n",
    "    \n",
    "    # Reformat HOMER output\n",
    "    output = outputDirectory + 'motifCounts.csv'\n",
    "    formattedTable = motifCountTable[ motifCountTable.columns[21:] ]\n",
    "    formattedTable = formattedTable.transpose()\n",
    "    formattedTable.columns = motifCountTable.loc[:,motifCountTable.columns[0]]\n",
    "    formattedTable.columns.name = 'PeakID'\n",
    "    formattedTable.to_csv(output,header=True, index=True, sep='\\t')\n",
    "    \n",
    "    print(\"Motif counts of the sequence have been placed in \" + output)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e76e37b",
   "metadata": {},
   "source": [
    "# getMotifCount( peakFile, genome, motifFile, outputDirectory (Optional) )\n",
    "This function provides the count of each motif in each peak.\n",
    "## Paramters:\n",
    "### peakFile\n",
    "File path to peak file \n",
    "### genome\n",
    "Genome that peak file is from \n",
    "### motifFile\n",
    ".motif file containing all the motifs\n",
    "### outputDirectory\n",
    "The output directory you would like the function to output to, this is an optional parameter, if not specified it will output in the directory where this code is run.\n",
    "Include the /, in your input, for example if the directory is glasslab/data, input it as glasslab/data/\n",
    "## Output\n",
    "The program will output the counts to a csv file named motifCounts.csv in the following format. \n",
    "The csv file is separated by tabs and has a header and index. \n",
    "You can read it in pandas like this pd.read_table('motifCounts.csv', sep='\\t', header=0, index_col=0)\n",
    "\n",
    "| File            | peak1 | peak2  |  ...  | peakX  |\n",
    "| --------------- | ----- | ------ | ----- | ------ |\n",
    "| Count of Motif1 |   a1  |   b1   |  ...  |   x1   |\n",
    "| Count of Motif2 |   a2  |   b2   |  ...  |   x2   |\n",
    "|  ...            |  ...  |   ...  |  ...  |  ...   |\n",
    "| Count of MotifY |   aY  |   bY   |  ...  |   xy   |"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "ab58e451",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl peaks_categ1.csv mm10 -size 500 -cpu 10 -m combinedMotifs.motif-nmotifs > motifCount.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPeak file = peaks_categ1.csv\n",
      "\tGenome = mm10\n",
      "\tOrganism = mouse\n",
      "\tPeak Region set to 500\n",
      "\tWill use up to 10 CPUs in parts that can use them\n",
      "\tMotif files:\n",
      "\t\tcombinedMotifs.motif\t-m\n",
      "\tWill report the number of motifs in each peak\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 0\n",
      "\t\tpeakfile formatted lines: 16093\n",
      "\t\tDuplicated Peak IDs: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 16093\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tResizing peaks...\n",
      "\tExtracting Sequence...\n",
      "\tCustom genome sequence directory: /home/glasslab/HOMER/.//data/genomes/mm10/\n",
      "\n",
      "\tExtracting sequences from directory: /home/glasslab/HOMER/.//data/genomes/mm10/\n",
      "\tExtracting 1133 sequences from chr1\n",
      "\tExtracting 1416 sequences from chr2\n",
      "\tExtracting 702 sequences from chr3\n",
      "\tExtracting 1003 sequences from chr4\n",
      "\tExtracting 1003 sequences from chr5\n",
      "\tExtracting 841 sequences from chr6\n",
      "\tExtracting 1109 sequences from chr7\n",
      "\tExtracting 781 sequences from chr8\n",
      "\tExtracting 844 sequences from chr9\n",
      "\tExtracting 937 sequences from chr10\n",
      "\tExtracting 1444 sequences from chr11\n",
      "\tExtracting 584 sequences from chr12\n",
      "\tExtracting 720 sequences from chr13\n",
      "\tExtracting 601 sequences from chr14\n",
      "\tExtracting 709 sequences from chr15\n",
      "\tExtracting 576 sequences from chr16\n",
      "\tExtracting 730 sequences from chr17\n",
      "\tExtracting 396 sequences from chr18\n",
      "\tExtracting 429 sequences from chr19\n",
      "\tExtracting 124 sequences from chrX\n",
      "\tExtracting 5 sequences from chrY\n",
      "\tExtracting 1 sequences from chr4_GL456216_random\n",
      "\tExtracting 5 sequences from chrUn_JH584304\n",
      "\tReading Positions...\n",
      "\tCalculating CpG/GC Content of Peaks\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 501 bp\n",
      "\t\t16093 total\n",
      "\n",
      "\t-----------------------\n",
      "\tLooking for Motifs...\n",
      "\n",
      "\tReading input files...\n",
      "\t16093 total sequences read\n",
      "\t3 motifs loaded\n",
      "\tFinding instances of 3 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tFinding Closest TSS...\n",
      "\tAnnotating:.......................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t226.0\t20759660\t0.881\t-37.882\n",
      "\t\tmiRNA\t0.0\t31126\t-0.244\t0.184\n",
      "\t\tncRNA\t35.0\t3444819\t0.781\t-6.224\n",
      "\t\tTTS\t279.0\t27313026\t0.789\t-38.340\n",
      "\t\tpseudo\t4.0\t548702\t0.302\t-0.898\n",
      "\t\tExon\t169.0\t34638742\t-0.277\t5.220\n",
      "\t\tIntron\t8277.0\t949839698\t0.559\t-924.015\n",
      "\t\tIntergenic\t5397.0\t1652327179\t-0.856\t2425.106\n",
      "\t\tPromoter\t1644.0\t30201318\t3.202\t-2258.037\n",
      "\t\t5UTR\t61.0\t2375227\t2.119\t-45.416\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\trRNA\t1.0\t5631\t4.908\t-3.419\n",
      "\tNOTE: If this part takes more than 2 minutes, there is a good chance\n",
      "\t\tyour machine ran out of memory: consider hitting ctrl+C and rerunning\n",
      "\t\tthe command with \"-noann\"\n",
      "\tTo capture annotation stats in a file, use \"-annStats <filename>\" next time\n",
      "\tAnnotating:.......................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t226.0\t20759660\t0.881\t-37.932\n",
      "\t\tOther\t9.0\t7949755\t-2.384\t24.979\n",
      "\t\tRC?\t0.0\t10979\t-0.091\t0.065\n",
      "\t\tRNA\t0.0\t113962\t-0.743\t0.674\n",
      "\t\tmiRNA\t0.0\t31126\t-0.244\t0.184\n",
      "\t\tncRNA\t35.0\t3444819\t0.782\t-6.231\n",
      "\t\tTTS\t279.0\t27313026\t0.789\t-38.397\n",
      "\t\tLINE\t766.0\t543499314\t-2.068\t1571.101\n",
      "\t\tLINE?\t0.0\t8168\t-0.068\t0.048\n",
      "\t\tsrpRNA\t1.0\t43307\t1.966\t-1.488\n",
      "\t\tSINE\t557.0\t196385695\t-1.059\t210.256\n",
      "\t\tRC\t0.0\t65629\t-0.473\t0.388\n",
      "\t\ttRNA\t8.0\t266421\t2.345\t-8.360\n",
      "\t\tDNA?\t0.0\t142070\t-0.880\t0.840\n",
      "\t\tpseudo\t4.0\t548702\t0.303\t-0.899\n",
      "\t\tDNA\t199.0\t28672218\t0.232\t-4.263\n",
      "\t\tExon\t169.0\t34638742\t-0.277\t5.202\n",
      "\t\tIntron\t6784.0\t602768002\t0.929\t-1607.471\n",
      "\t\tIntergenic\t4077.0\t822398691\t-0.254\t97.031\n",
      "\t\tPromoter\t1644.0\t30201318\t3.203\t-2258.735\n",
      "\t\t5UTR\t60.0\t2375227\t2.096\t-43.962\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\tLTR?\t3.0\t192563\t1.398\t-2.231\n",
      "\t\tscRNA\t5.0\t602434\t0.490\t-1.252\n",
      "\t\tCpG-Island\t63.0\t3121814\t1.772\t-35.530\n",
      "\t\tLow_complexity\t188.0\t19384972\t0.714\t-22.462\n",
      "\t\tLTR\t905.0\t312904556\t-1.031\t331.951\n",
      "\t\tSimple_repeat\t93.0\t57516916\t-1.870\t131.204\n",
      "\t\tsnRNA\t7.0\t237223\t2.320\t-7.374\n",
      "\t\tUnknown\t8.0\t2405068\t-0.829\t2.887\n",
      "\t\tSINE?\t0.0\t29758\t-0.234\t0.176\n",
      "\t\tSatellite\t2.0\t4567569\t-3.755\t21.044\n",
      "\t\trRNA\t1.0\t166488\t0.023\t-0.468\n",
      "\tCounting Tags in Peaks from each directory...\n",
      "\tOrganism: mouse\n",
      "\tLoading Gene Informaiton...\n",
      "\tOutputing Annotation File...\n",
      "\tDone annotating peaks file\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr12-1</th>\n",
       "      <th>chr17-1</th>\n",
       "      <th>chr12-2</th>\n",
       "      <th>chr14-1</th>\n",
       "      <th>chr11-2</th>\n",
       "      <th>chr12-3</th>\n",
       "      <th>chr8-1</th>\n",
       "      <th>chr11-1</th>\n",
       "      <th>chr7-1</th>\n",
       "      <th>chr2-4</th>\n",
       "      <th>...</th>\n",
       "      <th>chr15-2663</th>\n",
       "      <th>chr17-1899</th>\n",
       "      <th>chr10-3531</th>\n",
       "      <th>chr9-2505</th>\n",
       "      <th>chr15-3006</th>\n",
       "      <th>chr10-4479</th>\n",
       "      <th>chr3-4343</th>\n",
       "      <th>chr3-3737</th>\n",
       "      <th>chr11-7234</th>\n",
       "      <th>chr5-2985</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 16093 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                  chr12-1  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)        1   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)           1   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)         1   \n",
       "\n",
       "                                                                                                  chr17-1  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)        0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)           1   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)         0   \n",
       "\n",
       "                                                                                                  chr12-2  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)        2   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)           1   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)         1   \n",
       "\n",
       "                                                                                                  chr14-1  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)        2   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)           2   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)         1   \n",
       "\n",
       "                                                                                                  chr11-2  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)        0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)         0   \n",
       "\n",
       "                                                                                                  chr12-3  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)        0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)           1   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)         1   \n",
       "\n",
       "                                                                                                  chr8-1  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)       2   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)          2   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)        3   \n",
       "\n",
       "                                                                                                  chr11-1  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)        0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)           2   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)         0   \n",
       "\n",
       "                                                                                                  chr7-1  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)       1   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)          2   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)        0   \n",
       "\n",
       "                                                                                                  chr2-4  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)       0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)          1   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)        0   \n",
       "\n",
       "                                                                                                  ...  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)  ...   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     ...   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)   ...   \n",
       "\n",
       "                                                                                                  chr15-2663  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)           1   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              1   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr17-1899  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr10-3531  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr9-2505  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)             0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "\n",
       "                                                                                                  chr15-3006  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)           1   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr10-4479  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)           1   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              2   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr3-4343  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)             0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "\n",
       "                                                                                                  chr3-3737  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)             0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "\n",
       "                                                                                                  chr11-7234  \\\n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr5-2985  \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)          0  \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)             0  \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)           0  \n",
       "\n",
       "[3 rows x 16093 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "getMotifCount(\"peaks_categ1.csv\",\"mm10 -size 500 -cpu 10\",\"combinedMotifs.motif\")\n",
    "COUNTS = pd.read_table('motifCounts.csv', sep='\\t', header=0, index_col=0)\n",
    "COUNTS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "68071cbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "def getMotifLocation(peakFile, genome, motifFile, outputDirectory=\"\", motifCountFile=\"\"):\n",
    "    \n",
    "    if not os.path.exists(peakFile):\n",
    "        print('Path to peak file is incorrect or does not exist\\n')\n",
    "        return 1\n",
    "\n",
    "    if not os.path.exists(motifFile):\n",
    "        print('Path to motifs is incorrect or does not exist\\n')\n",
    "        return 1\n",
    "    \n",
    "    ## Read in motifFile containing path to .motif files\n",
    "    motifs = pd.read_table(motifFile, sep=\"\\n\", header=None)\n",
    "    motifs = motifs.to_numpy()\n",
    "    \n",
    "    ## Create a string combinining all the .motif paths\n",
    "#     motifString = \"\"\n",
    "#     numOfMotif = motifs.size\n",
    "#     for i in motifs:\n",
    "#         motifString = motifString + i[0] + \" \"\n",
    "        \n",
    "    ## Output File\n",
    "    homerOutput = outputDirectory + \"motifLocations.txt\"\n",
    "    \n",
    "    # Get Motif Counts\n",
    "    motifCounts = None\n",
    "    if motifCountFile == \"\":\n",
    "        getMotifCount(peakFile, genome, motifFile, outputDirectory)\n",
    "        motifCounts = pd.read_table(str(outputDirectory + \"motifCounts.csv\"), sep=\"\\t\", header=0, index_col=0)\n",
    "    else:\n",
    "        motifCounts = pd.read_table(motifCountFile, sep=\"\\t\", index_col=0)\n",
    "    \n",
    "    ## Call HOMER annotatePeaks to get motif positions\n",
    "    print('annotatePeaks.pl ' + peakFile + ' ' + genome + ' -m ' + motifFile + ' > '+ homerOutput)\n",
    "    os.system('annotatePeaks.pl ' + peakFile + ' ' + genome + ' -m ' + motifFile + ' > '+ homerOutput)\n",
    "    \n",
    "    motifLocations = pd.read_table(homerOutput)\n",
    "    \n",
    "    # Set up Columns\n",
    "    columnNames = motifLocations.columns[21:]\n",
    "    columnNames = columnNames.insert(0, motifLocations.columns[0])\n",
    "    columnNames = columnNames.insert(1, motifLocations.columns[1])\n",
    "    columnNames = columnNames.insert(2, 'Start')\n",
    "    columnNames = columnNames.insert(3, 'End')\n",
    "    \n",
    "    # Set up position Table\n",
    "    positionTable = motifLocations[ columnNames ]\n",
    "    positionTable.to_csv('positionTable.csv',header=True, index=True, sep='\\t')\n",
    "    positionTable = positionTable.transpose()\n",
    "    positionTable.columns = motifLocations[motifLocations.columns[0]]\n",
    "    positionTable.columns.name = \"PeakID\"\n",
    "    \n",
    "    # Starting X and Ending X for Peak\n",
    "    peakStart = positionTable.iloc[2]\n",
    "    peakEnd = positionTable.iloc[3]\n",
    "    peakMid = (peakEnd-peakStart)/2+peakStart\n",
    "    \n",
    "    #\n",
    "    finalMotifCoords = []\n",
    "    for i in range(positionTable.columns.size):\n",
    "        peakMotifCoord = []\n",
    "        peakID = positionTable.columns[i]\n",
    "        motifCountsForPeak = motifCounts[positionTable.columns[i]]\n",
    "        for j in range(motifCountsForPeak.shape[0]):\n",
    "            peakMotifCount = motifCountsForPeak[j]\n",
    "            if peakMotifCount == 0:\n",
    "                peakMotifCoord.append(0)\n",
    "            else:\n",
    "                motifCoords = positionTable.loc[motifCounts.index[j]][peakID].split(\" \")\n",
    "                # Coordinate Format = Offset(Sequence,-/+,Score)\n",
    "                for k in motifCoords:\n",
    "                    motifFormat = k.split(\"(\")\n",
    "                    offset = int(motifFormat[0])\n",
    "                    if k == motifCoords[0]:\n",
    "                        peakMotifCoord.append(int(peakMid[peakID])+int(offset))\n",
    "        finalMotifCoords.append(peakMotifCoord)\n",
    "\n",
    "    finalMotifCoords = pd.DataFrame(finalMotifCoords)\n",
    "    finalMotifCoords.columns = [motifCounts.index]\n",
    "    finalMotifCoords.index = [positionTable.columns]\n",
    "    finalMotifCoords = finalMotifCoords.transpose()\n",
    "    \n",
    "    output = outputDirectory + 'motifCoordinates.csv'\n",
    "    finalMotifCoords.to_csv(output,header=True, index=True, sep='\\t')\n",
    "    \n",
    "    print(\"Motif Coordinates of the peak file have been placed in \" + output)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "278718bd",
   "metadata": {},
   "source": [
    "# getMotifLocation( peakFile, genome, motifFile, outputDirectory (Optional), motifCount (Optional))\n",
    "This function provides the starting location of motifs in each peak, if there is more than one motif in a peak, it will provide the starting location of the first motif.\n",
    "## Paramters:\n",
    "### peakFile\n",
    "File path to peak file \n",
    "### genome\n",
    "Genome that peak file is from \n",
    "### motifFile\n",
    ".motif file containing all the motifs\n",
    "### outputDirectory\n",
    "The output directory you would like the function to output to, this is an optional parameter, if not specified it will output in the directory where this code is run.\n",
    "Include the /, in your input, for example if the directory is glasslab/data, input it as glasslab/data/\n",
    "### motifCounts\n",
    "csv file containing the counts of motif, only the format outputted by getMotifCounts will work. This is an optional parameter\n",
    "\n",
    "## Output\n",
    "The program will output the coordinates to a csv file named motifCoordinates.csv in the following format. \n",
    "The csv file is separated by tabs and has a header and index. \n",
    "You can read it in pandas like this pd.read_table('motifCoordinates.csv', sep='\\t', header=0, index_col=0)\n",
    "\n",
    "| File            | peak1 | peak2  |  ...  | peakX  |\n",
    "| --------------- | ----- | ------ | ----- | ------ |\n",
    "| Starting Coordinate of Motif1 for Corresponding Peak: |   a1  |   b1   |  ...  |   x1   |\n",
    "| Starting Coordinate of Motif2 for Corresponding Peak: |   a2  |   b2   |  ...  |   x2   |\n",
    "|  ...            |  ...  |   ...  |  ...  |  ...   |\n",
    "| Starting Coordinate of MotifY for Corresponding Peak: |   aY  |   bY   |  ...  |   xy   |"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "488ba408",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl peaks_categ1.csv mm10 -size 500 -cpu 10 -m combinedMotifs.motif > motifLocations.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPeak file = peaks_categ1.csv\n",
      "\tGenome = mm10\n",
      "\tOrganism = mouse\n",
      "\tPeak Region set to 500\n",
      "\tWill use up to 10 CPUs in parts that can use them\n",
      "\tMotif files:\n",
      "\t\tcombinedMotifs.motif\t-m\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 0\n",
      "\t\tpeakfile formatted lines: 16093\n",
      "\t\tDuplicated Peak IDs: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 16093\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tResizing peaks...\n",
      "\tExtracting Sequence...\n",
      "\tCustom genome sequence directory: /home/glasslab/HOMER/.//data/genomes/mm10/\n",
      "\n",
      "\tExtracting sequences from directory: /home/glasslab/HOMER/.//data/genomes/mm10/\n",
      "\tExtracting 1133 sequences from chr1\n",
      "\tExtracting 1416 sequences from chr2\n",
      "\tExtracting 702 sequences from chr3\n",
      "\tExtracting 1003 sequences from chr4\n",
      "\tExtracting 1003 sequences from chr5\n",
      "\tExtracting 841 sequences from chr6\n",
      "\tExtracting 1109 sequences from chr7\n",
      "\tExtracting 781 sequences from chr8\n",
      "\tExtracting 844 sequences from chr9\n",
      "\tExtracting 937 sequences from chr10\n",
      "\tExtracting 1444 sequences from chr11\n",
      "\tExtracting 584 sequences from chr12\n",
      "\tExtracting 720 sequences from chr13\n",
      "\tExtracting 601 sequences from chr14\n",
      "\tExtracting 709 sequences from chr15\n",
      "\tExtracting 576 sequences from chr16\n",
      "\tExtracting 730 sequences from chr17\n",
      "\tExtracting 396 sequences from chr18\n",
      "\tExtracting 429 sequences from chr19\n",
      "\tExtracting 124 sequences from chrX\n",
      "\tExtracting 5 sequences from chrY\n",
      "\tExtracting 1 sequences from chr4_GL456216_random\n",
      "\tExtracting 5 sequences from chrUn_JH584304\n",
      "\tReading Positions...\n",
      "\tCalculating CpG/GC Content of Peaks\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 501 bp\n",
      "\t\t16093 total\n",
      "\n",
      "\t-----------------------\n",
      "\tLooking for Motifs...\n",
      "\n",
      "\tReading input files...\n",
      "\t16093 total sequences read\n",
      "\t3 motifs loaded\n",
      "\tFinding instances of 3 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tFinding Closest TSS...\n",
      "\tAnnotating:.......................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t226.0\t20759660\t0.881\t-37.882\n",
      "\t\tmiRNA\t0.0\t31126\t-0.244\t0.184\n",
      "\t\tncRNA\t35.0\t3444819\t0.781\t-6.224\n",
      "\t\tTTS\t279.0\t27313026\t0.789\t-38.340\n",
      "\t\tpseudo\t4.0\t548702\t0.302\t-0.898\n",
      "\t\tExon\t169.0\t34638742\t-0.277\t5.220\n",
      "\t\tIntron\t8277.0\t949839698\t0.559\t-924.015\n",
      "\t\tIntergenic\t5397.0\t1652327179\t-0.856\t2425.106\n",
      "\t\tPromoter\t1644.0\t30201318\t3.202\t-2258.037\n",
      "\t\t5UTR\t61.0\t2375227\t2.119\t-45.416\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\trRNA\t1.0\t5631\t4.908\t-3.419\n",
      "\tNOTE: If this part takes more than 2 minutes, there is a good chance\n",
      "\t\tyour machine ran out of memory: consider hitting ctrl+C and rerunning\n",
      "\t\tthe command with \"-noann\"\n",
      "\tTo capture annotation stats in a file, use \"-annStats <filename>\" next time\n",
      "\tAnnotating:.......................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t226.0\t20759660\t0.881\t-37.932\n",
      "\t\tOther\t9.0\t7949755\t-2.384\t24.979\n",
      "\t\tRC?\t0.0\t10979\t-0.091\t0.065\n",
      "\t\tRNA\t0.0\t113962\t-0.743\t0.674\n",
      "\t\tmiRNA\t0.0\t31126\t-0.244\t0.184\n",
      "\t\tncRNA\t35.0\t3444819\t0.782\t-6.231\n",
      "\t\tTTS\t279.0\t27313026\t0.789\t-38.397\n",
      "\t\tLINE\t766.0\t543499314\t-2.068\t1571.101\n",
      "\t\tLINE?\t0.0\t8168\t-0.068\t0.048\n",
      "\t\tsrpRNA\t1.0\t43307\t1.966\t-1.488\n",
      "\t\tSINE\t557.0\t196385695\t-1.059\t210.256\n",
      "\t\tRC\t0.0\t65629\t-0.473\t0.388\n",
      "\t\ttRNA\t8.0\t266421\t2.345\t-8.360\n",
      "\t\tDNA?\t0.0\t142070\t-0.880\t0.840\n",
      "\t\tpseudo\t4.0\t548702\t0.303\t-0.899\n",
      "\t\tDNA\t199.0\t28672218\t0.232\t-4.263\n",
      "\t\tExon\t169.0\t34638742\t-0.277\t5.202\n",
      "\t\tIntron\t6784.0\t602768002\t0.929\t-1607.471\n",
      "\t\tIntergenic\t4077.0\t822398691\t-0.254\t97.031\n",
      "\t\tPromoter\t1644.0\t30201318\t3.203\t-2258.735\n",
      "\t\t5UTR\t60.0\t2375227\t2.096\t-43.962\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\tLTR?\t3.0\t192563\t1.398\t-2.231\n",
      "\t\tscRNA\t5.0\t602434\t0.490\t-1.252\n",
      "\t\tCpG-Island\t63.0\t3121814\t1.772\t-35.530\n",
      "\t\tLow_complexity\t188.0\t19384972\t0.714\t-22.462\n",
      "\t\tLTR\t905.0\t312904556\t-1.031\t331.951\n",
      "\t\tSimple_repeat\t93.0\t57516916\t-1.870\t131.204\n",
      "\t\tsnRNA\t7.0\t237223\t2.320\t-7.374\n",
      "\t\tUnknown\t8.0\t2405068\t-0.829\t2.887\n",
      "\t\tSINE?\t0.0\t29758\t-0.234\t0.176\n",
      "\t\tSatellite\t2.0\t4567569\t-3.755\t21.044\n",
      "\t\trRNA\t1.0\t166488\t0.023\t-0.468\n",
      "\tCounting Tags in Peaks from each directory...\n",
      "\tOrganism: mouse\n",
      "\tLoading Gene Informaiton...\n",
      "\tOutputing Annotation File...\n",
      "\tDone annotating peaks file\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr12-1</th>\n",
       "      <th>chr17-1</th>\n",
       "      <th>chr12-2</th>\n",
       "      <th>chr14-1</th>\n",
       "      <th>chr11-2</th>\n",
       "      <th>chr12-3</th>\n",
       "      <th>chr8-1</th>\n",
       "      <th>chr11-1</th>\n",
       "      <th>chr7-1</th>\n",
       "      <th>chr2-4</th>\n",
       "      <th>...</th>\n",
       "      <th>chr15-2663</th>\n",
       "      <th>chr17-1899</th>\n",
       "      <th>chr10-3531</th>\n",
       "      <th>chr9-2505</th>\n",
       "      <th>chr15-3006</th>\n",
       "      <th>chr10-4479</th>\n",
       "      <th>chr3-4343</th>\n",
       "      <th>chr3-3737</th>\n",
       "      <th>chr11-7234</th>\n",
       "      <th>chr5-2985</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PeakID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)</th>\n",
       "      <td>83929581</td>\n",
       "      <td>0</td>\n",
       "      <td>102451552</td>\n",
       "      <td>73222678</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56313776</td>\n",
       "      <td>0</td>\n",
       "      <td>55852957</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>35917582</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>102234303</td>\n",
       "      <td>86778582</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)</th>\n",
       "      <td>83929580</td>\n",
       "      <td>88551596</td>\n",
       "      <td>102451744</td>\n",
       "      <td>73222739</td>\n",
       "      <td>0</td>\n",
       "      <td>52397933</td>\n",
       "      <td>56313775</td>\n",
       "      <td>110166021</td>\n",
       "      <td>55852958</td>\n",
       "      <td>16335701</td>\n",
       "      <td>...</td>\n",
       "      <td>35917583</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>86778478</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)</th>\n",
       "      <td>83929351</td>\n",
       "      <td>0</td>\n",
       "      <td>102451664</td>\n",
       "      <td>73222676</td>\n",
       "      <td>0</td>\n",
       "      <td>52397932</td>\n",
       "      <td>56313774</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 16093 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                   chr12-1  \\\n",
       "PeakID                                                                                                       \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)  83929581   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     83929580   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)   83929351   \n",
       "\n",
       "                                                                                                   chr17-1  \\\n",
       "PeakID                                                                                                       \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)         0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     88551596   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "\n",
       "                                                                                                    chr12-2  \\\n",
       "PeakID                                                                                                        \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)  102451552   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     102451744   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)   102451664   \n",
       "\n",
       "                                                                                                   chr14-1  \\\n",
       "PeakID                                                                                                       \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)  73222678   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     73222739   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)   73222676   \n",
       "\n",
       "                                                                                                  chr11-2  \\\n",
       "PeakID                                                                                                      \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)        0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)         0   \n",
       "\n",
       "                                                                                                   chr12-3  \\\n",
       "PeakID                                                                                                       \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)         0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     52397933   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)   52397932   \n",
       "\n",
       "                                                                                                    chr8-1  \\\n",
       "PeakID                                                                                                       \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)  56313776   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     56313775   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)   56313774   \n",
       "\n",
       "                                                                                                    chr11-1  \\\n",
       "PeakID                                                                                                        \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     110166021   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "\n",
       "                                                                                                    chr7-1  \\\n",
       "PeakID                                                                                                       \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)  55852957   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     55852958   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "\n",
       "                                                                                                    chr2-4  \\\n",
       "PeakID                                                                                                       \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)         0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     16335701   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "\n",
       "                                                                                                  ...  \\\n",
       "PeakID                                                                                            ...   \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)  ...   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)     ...   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)   ...   \n",
       "\n",
       "                                                                                                  chr15-2663  \\\n",
       "PeakID                                                                                                         \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)    35917582   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)       35917583   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr17-1899  \\\n",
       "PeakID                                                                                                         \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr10-3531  \\\n",
       "PeakID                                                                                                         \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr9-2505  \\\n",
       "PeakID                                                                                                        \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)             0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "\n",
       "                                                                                                  chr15-3006  \\\n",
       "PeakID                                                                                                         \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)   102234303   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr10-4479  \\\n",
       "PeakID                                                                                                         \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)    86778582   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)       86778478   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr3-4343  \\\n",
       "PeakID                                                                                                        \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)             0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "\n",
       "                                                                                                  chr3-3737  \\\n",
       "PeakID                                                                                                        \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)          0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)             0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "\n",
       "                                                                                                  chr11-7234  \\\n",
       "PeakID                                                                                                         \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)           0   \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)              0   \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)            0   \n",
       "\n",
       "                                                                                                  chr5-2985  \n",
       "PeakID                                                                                                       \n",
       "PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer Distance From Peak(sequence,strand,conservation)          0  \n",
       "Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer Distance From Peak(sequence,strand,conservation)             0  \n",
       "SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer Distance From Peak(sequence,strand,conservation)           0  \n",
       "\n",
       "[3 rows x 16093 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "getMotifLocation(\"peaks_categ1.csv\",\"mm10 -size 500 -cpu 10\",\"combinedMotifs.motif\",\"\", \"motifCounts.csv\")\n",
    "COORDS = pd.read_table('motifCoordinates.csv', sep='\\t', header=0, index_col=0)\n",
    "COORDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff8ed479",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
